library/src/main/kotlin/org/codice/compliance/Common.kt

```diff
@@ -125,11 +125,15 @@ fun Node.allChildren(name: String): List<Node> {
     return nodes
 }
 
-fun Node.prettyPrintXml(): String {
-    val transformer = createTransformer()
-    val output = StringWriter()
-    transformer.transform(DOMSource(this), StreamResult(output))
-    return output.toString()
+fun Node.prettyPrintXmlOnDebug(): String? {
+    if (Log.logLevel == LogLevel.DEBUG) {
+        val transformer = createTransformer()
+        val output = StringWriter()
+        transformer.transform(DOMSource(this), StreamResult(output))
+        return output.toString()
+    } else {
+        return null
+    }
```

Comment:
Changed this because, when getting lots of exceptions, the logs were basically unreadable.